packages/foundation-models/src/openai/model-types.ts

```diff
@@ -0,0 +1,14 @@
+import type {
+  AzureOpenAiChatModel as CoreAzureOpenAiChatModel,
+  AzureOpenAiEmbeddingModel as CoreAzureOpenAiEmbeddingModel
+} from '@sap-ai-sdk/core';
+
+/**
+ * Azure OpenAI models for chat completion.
+ */
+export type AzureOpenAiChatModel = CoreAzureOpenAiChatModel;
```

Comment:
[q] why the importing and re-exporting?

Reply:
True, I can also not think of a case, where a model defined in `core` could be unavailable in `foundation-models`. Adjusted.

Reply:
I remember in the Cloud SDK we decided types in odata-common that were used odata-v4/v2 should be re-exported from v2/v3 since we dont want users to import odata-common package. 

Shouldn't we do the same here? @marikaner @tomfrenken opinions?

Reply:
Oh now I am again upvote for re-exporting as `core` is not published anyways. They will not be able to import from `core`.

Reply:
@ZhongpinWang core is published!

I don't have a very strong opinion on this, but @deekshas8 just pointed out that there are other types we should then include (e.g. `ModelDeployment` and co.).
I am leaning towards not re-exporting this (we can always do it later if needed), because these types here as well as `ModelDeployment` and similar are all types that users will use in their API, but most likely won't need explicitly.
In any case: whatever we do it should be consistent.

Reply:
@marikaner Okay. I read the [README.md](https://github.com/SAP/ai-sdk-js/tree/main/packages/core#sap-ai-sdkcore) from core and thought since it is primarily for internal use, we would rather want to hide this package. `core` is not even in the main README.

Regarding consistency, these are what we are currently exporting in `core` index:

```ts
export type {
  BaseLlmParametersWithDeploymentId,
  BaseLlmParameters,
  CustomRequestConfig,
  EndpointOptions
} from './http-client.js';
export { executeRequest } from './http-client.js';
export { getAiCoreDestination } from './context.js';
export { OpenApiRequestBuilder } from './openapi-request-builder.js';
export type {
  AzureOpenAiChatModel,
  AzureOpenAiEmbeddingModel,
  GcpVertexAiChatModel,
  AwsBedrockChatModel,
  ChatModel
} from './model-types.js';
```

Anything we should directly expose in other packages other than `core`?

Reply:
There is nothing that we would want users to use directly. Except maybe the Models, that is right. `ModelDeployment` and co is on `ai-api` not `core`. Honestly, I don't have a strong opinion. Semantically it belongs into `foundation-models` anyways, we only need it in core to be able to reuse it in orchestration. `@types/sap-ai-sdk_foundation-models` would be better...

Reply:
We could consider this, but it might be total overkill: https://github.com/DefinitelyTyped/DefinitelyTyped

Reply:
Or a separate package `@sap-ai-sdk/foundation-model-types`. Might also be overkill

Reply:
> There is nothing that we would want users to use directly. Except maybe the Models, that is right. `ModelDeployment` and co is on `ai-api` not `core`. Honestly, I don't have a strong opinion. Semantically it belongs into `foundation-models` anyways, we only need it in core to be able to reuse it in orchestration. `@types/sap-ai-sdk_foundation-models` would be better...

There is another reason why I put it into core, because `foundation-models` currently do not support all models and so that it won't confuse the user, I put it in core and then only pick what I need from `foundation-models`. Otherwise, `orchestration` can also import models from `foundation-models`. I now treat `core` as some kind of common library. 

I don't think re-exporting is a bad idea, just in case in the future, we move again models from `core` back to `foundation-models`, we just have to put models into this `model-types.ts` in `foundation-models`. Nothing else needs to be changed except we remove it from `core`, which should probably not be used directly from a user.